website/content/api-docs/secret/openldap.mdx

```diff
@@ -216,36 +240,274 @@ $ curl \
+may be: `v_test-role_1234567890`. Note the `-` (dash) in the `test-role`. If the LDAP system Vault is managing
+restricts usernames/DNs to not allow for dashes (or other characters), the template must explicitly modify/remove those
+characters. In this example, the template can be changed to `v_{{.RoleName | replace "-" "_"}}` to replace
+the dashes with underscores. See [Template Functions](#template-functions) for more functions available.
+
+`.RoleName` - The name of the role the credentials are being generated from.
+
+`.DisplayName` - The display name associated with the user making the request against Value.
+
+#### Template Functions
+
+Both the LDIF templates and the username template use the [Go template language](https://golang.org/pkg/text/template)
+so all [functions](https://golang.org/pkg/text/template/#hdr-Functions) and capabilities from that language are
+supported including functions such as `printf`.
+
+In addition to the functions available through the template language, the following custom functions are also available:
+
+`random` - Generates a random string from lowercase letters, uppercase letters, and numbers. Must include a
+number indicating how many characters to generate.<br />
+**Example:** `{{random 20}}` generates 20 random characters
+
+`truncate` - truncates the previous value to the provided number of characters.<br />
+**Example:** `{{.FieldName | truncate 10}}`
+
+`truncate_sha256` - Truncates the previous value to the provided number of characters. The last 8 characters of the
+truncated value will be the first 8 characters of the SHA256 hash of the truncated characters.
+
+**Example:** `{{.FieldName | truncate_sha256 20}}`.<br />
+If `FieldName` is `abcdefghijklmnopqrstuvwxyz`, all characters after the 12th (`l`) are removed and SHA256 hashed
+(`872808ffbf0158b200d397a9884342497f23b153b1630229014cea1886ca6f20`). The first 8 characters of the hash (`872808ff`)
+are then appended to the end of the first 12 characters from the original value: `abcdefghijkl872808ff`.
+
+This can aid in identifying values (such as the role & display names) if they need to be truncated to a particular
+length, particularly if the value has common prefixes. An example of this is having many roles with a common prefix in
+the name of the role, but the role name is truncated such that only the prefix is shown. This function will keep the
+non-prefix part of the role name in the username which will aid in identification while also keeping uniqueness.
+
+**Example Scenario:**<br />
+Two roles exist: `mylongprefix-foobar` and `mylongprefix-bazqux`. If the template is
+`v_{{.RoleName | truncate 12}}_{{unix_time}}` the username would be `v_mylongprefix_1234567890` for both of the roles.
+However, if the template is `v_{{.RoleName | truncate_sha256 12}}_{{unix_time}}` the username for the first role
+would be: `v_mylo998a317d_1234567890` and the username for the second role would be: `v_mylo224e847c_1234567890`.
```

Comment:
The two examples can be combined into one to convey the reasoning.

```suggestion
This can aid in identifying values (such as the role & display names) if they need to be truncated to a particular
length, particularly if the value has common prefixes. An example of this is having many roles with a common prefix in
the name of the role, but the role name is truncated such that only the prefix is shown. This function will keep the
non-prefix part of the role name in the username which will aid in identification while also keeping uniqueness.

**Example:**<br />
`v_{{.RoleName | truncate_sha256 12}}_{{unix_time}}`. <br />
If two roles exist for this template, `mylongprefix-foobar` and `mylongprefix-bazqux`, the username for the first role
would be `v_mylo998a317d_1234567890` and the username for the second role would be `v_mylo224e847c_1234567890`.
```

Reply:
:+1: with some tiny formatting changes